adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/attendance/infrastructure/attendanceDatabaseConnector.kt

```diff
@@ -1,30 +1,74 @@
 package com.krzykrucz.elesson.currentlesson.attendance.infrastructure
 
+import arrow.core.andThen
+import arrow.core.toOption
+import arrow.data.OptionT
+import arrow.effects.ForIO
 import arrow.effects.IO
-import arrow.effects.extensions.io.functor.unit
-import com.krzykrucz.elesson.currentlesson.domain.attendance.CheckedAttendance
-import com.krzykrucz.elesson.currentlesson.domain.attendance.GetLessonStartTime
-import com.krzykrucz.elesson.currentlesson.domain.startlesson.ClassName
+import arrow.effects.extensions.io.applicative.applicative
+import com.krzykrucz.elesson.currentlesson.attendance.AttendanceResponseDto
+import com.krzykrucz.elesson.currentlesson.domain.attendance.*
 import com.krzykrucz.elesson.currentlesson.domain.startlesson.ClassRegistry
 import com.krzykrucz.elesson.currentlesson.domain.startlesson.LessonIdentifier
+import com.krzykrucz.elesson.currentlesson.domain.startlesson.StartedLesson
 import com.krzykrucz.elesson.currentlesson.infrastructure.Database.Companion.ATTENDANCE_DATABASE
-import com.krzykrucz.elesson.currentlesson.infrastructure.Database.Companion.CLASS_REGISTRY_DATABASE
-import com.krzykrucz.elesson.currentlesson.infrastructure.Database.Companion.SCHEDULE
+import com.krzykrucz.elesson.currentlesson.infrastructure.Database.Companion.STARTED_LESSON_DATABASE
 
-typealias FetchClassRegistry = (ClassName) -> IO<ClassRegistry>
-typealias PersistCheckedAttendance = (LessonIdentifier, CheckedAttendance) -> IO<Unit>
+typealias PersistAttendance = (Attendance) -> IO<AttendanceResponseDto>
+typealias FetchAttendance = (LessonIdentifier) -> OptionT<ForIO, Attendance>
+typealias FetchCheckedAttendance = (LessonIdentifier) -> IO<CheckedAttendance>
+typealias FetchStartedLesson = (LessonIdentifier) -> IO<StartedLesson>
+typealias FetchClassRegistry = (Attendance) -> IO<ClassRegistry>
+typealias FetchStartedLessonAsAttendance = (LessonIdentifier) -> IO<NotCompletedAttendance>
 
 
-fun fetchClassRegistry(): FetchClassRegistry = { className ->
-    IO.just(CLASS_REGISTRY_DATABASE[className]!!)
+fun fetchAttendance(): FetchAttendance = { lessonIdentifier ->
+    OptionT.fromOption(IO.applicative(), ATTENDANCE_DATABASE.get(lessonIdentifier).toOption())
 }
 
+fun fetchCheckedAttendance(): FetchCheckedAttendance = { lessonIdentifier ->
+    IO.just(ATTENDANCE_DATABASE.get(lessonIdentifier)!! as CheckedAttendance)
+}
+
+fun fetchStartedLesson(): FetchStartedLesson = { lessonIdentifier ->
+    IO.just(STARTED_LESSON_DATABASE.get(lessonIdentifier)!!)
```

Comment:
exception will be thrown here, I'd prefer an explicit error as a return type instead (like LessonNotStarted)

Reply:
Yeah, I assumed that in attendance context there always will be a started lesson, but ok, rewrote it to OptionT